hal/src/rtl872x/pwm_hal.cpp

```diff
@@ -182,21 +179,22 @@ int pwmPinDeinit(uint16_t pin) {
     const hal_pin_info_t* pinInfo = hal_pin_map() + pin;
     const uint8_t instance = pinInfo->pwm_instance;
     const uint8_t channel = pinInfo->pwm_channel;
+    const pwm_state_t channelState = pwmInfo[instance].channels[channel].state;
 
-    if (pwmInfo[instance].pins[channel] != PIN_INVALID) {
+    if (channelState == PWM_STATE_ENABLED) {
         const uint32_t rtlPin = hal_pin_to_rtl_pin(pin);
         Pinmux_Config(rtlPin, PINMUX_FUNCTION_GPIO);
 
-        pwmInfo[instance].pins[channel] = PIN_INVALID;

```

Comment:
Setting the pin to `PIN_INVALID` made it impossible to re-generate the PWM signal on the suspended PWM channel when `hal_pwm_sleep(true) / hal_pwm_sleep(false)` was called. Thats why I changed pwm hal to track each channel state. When a pin is init'd, the pin number will always be preserved